src/System.Drawing.Design.Primitives/src/System/Drawing/Design/ToolboxItem.cs

```diff
@@ -0,0 +1,967 @@
+        /// to change the value of an object before comitting it, our reject it by throwing an 
+        /// exception.
+        /// </summary>
+        protected virtual object ValidatePropertyValue(string propertyName, object value) {
+            switch (propertyName) {
+                case "AssemblyName":
+                    ValidatePropertyType(propertyName, value, typeof(AssemblyName), true);
+                    break;
+
+                case "Bitmap":
+                    ValidatePropertyType(propertyName, value, typeof(Bitmap), true);
+                    break;
+
+                case "OriginalBitmap":
+                    ValidatePropertyType(propertyName, value, typeof(Bitmap), true);
+                    break;
+
+                case "Company":
+                case "Description":
+                case "DisplayName":
+                case "TypeName":
+                    ValidatePropertyType(propertyName, value, typeof(string), true);
+                    if (value == null) value = string.Empty;
+
+                    break;
+
+                case "Filter":
+                    ValidatePropertyType(propertyName, value, typeof(ICollection), true);
+
+                    int filterCount = 0;
+                    ICollection col = (ICollection)value;
+
+                    if (col != null) {
+                        foreach (object f in col) {
+                            if (f is ToolboxItemFilterAttribute) {
+                                filterCount++;
+                            }
+                        }
+                    }
+
+                    ToolboxItemFilterAttribute[] filter = new ToolboxItemFilterAttribute[filterCount];
+
+                    if (col != null) {
+                        filterCount = 0;
+                        foreach (object f in col) {
+                            ToolboxItemFilterAttribute tfa = f as ToolboxItemFilterAttribute;
+                            if (tfa != null) {
+                                filter[filterCount++] = tfa;
+                            }
+                        }
+                    }
+
+                    value = filter;
+                    break;
+
+                case "IsTransient":
+                    ValidatePropertyType(propertyName, value, typeof(bool), false);
+                    break;
+            }
+            return value;
+        }
+
+        /// <include file='doc\ToolboxItem.uex' path='docs/doc[@for="ToolboxItem.ISerializable.GetObjectData"]/*' />
+        /// <internalonly/> 
+        // SECREVIEW NOTE: we do not put the linkdemand that should be here, because the one on the type is a superset of this one
+        [SuppressMessage("Microsoft.Usage", "CA2240:ImplementISerializableCorrectly")]        
+        void ISerializable.GetObjectData(SerializationInfo info, StreamingContext context) {
+            var UnmanagedCode = new SecurityPermission(SecurityPermissionFlag.UnmanagedCode);
+            UnmanagedCode.Demand();
```

Comment:
CAS isn't needed.

Reply:
This need to be deleted, we don't need CAS in .NET Core anymore.